linkml/generators/prefixmapgen.py

```diff
@@ -64,11 +66,32 @@ def end_schema(self, base: Optional[str] = None, output: Optional[str] = None, *
             context[k] = v
         for k, v in self.slot_class_maps.items():
             context[k] = v
+
         if output:
-            with open(output, 'w') as outf:
-                outf.write(as_json(context))
+            output_ext = output.split(".")[-1]
```

Comment:
We should probably warn if this over-rides the users explicit choice

Reply:
Hm, so this isn't necessarily overriding anything. It just parses out the file extension from the filename that is supplied via the `--output` option.